api/v1beta1/dbinstance_webhook.go

```diff
@@ -75,6 +81,37 @@ func (r *DbInstance) ValidateUpdate(old runtime.Object) (admission.Warnings, err
 	return nil, nil
 }
 
+func ValidateConfigVsConfigFrom(r *GenericInstance) error {
+	if r != nil {
+		if len(r.Host) > 0 && r.HostFrom != nil {
+			return errors.New("it's not allowed to use both host and hostFrom, please choose one")
+		}
+		if len(r.PublicIP) > 0 && r.PublicIPFrom != nil {
+			return errors.New("it's not allowed to use both publicIp and publicIpFrom, please choose one")
+		}
+		if r.Port > 0 && r.PortFrom != nil {
+			return errors.New("it's not allowed to use both port and portFrom, please choose one")
+		}
+	}
+	return nil
+}
+
+func ValidateConfigFrom(dbin *GenericInstance) error {
+	check := dbin.HostFrom
+	if check != nil && !(check.Kind == kube.CONFIGMAP || check.Kind == kube.SECRET) {
+		return fmt.Errorf("unsupported kind in hostFrom: %s", check.Kind)
+	}
+	check = dbin.PortFrom
+	if check != nil && !(check.Kind == kube.CONFIGMAP || check.Kind == kube.SECRET) {
+		return fmt.Errorf("unsupported kind in portFrom: %s", check.Kind)
+	}
+	check = dbin.PublicIPFrom
+	if check != nil && !(check.Kind == kube.CONFIGMAP || check.Kind == kube.SECRET) {
+		return fmt.Errorf("unsupported kind in publicIpFrom: %s", check.Kind)
```

Comment:
Also here more friendly error message:

`Unsupported source kind %s in publicIpFrom, please use "configmap" or "secret"`